lib/n_plus_one_control/rspec/matcher.rb

```diff
@@ -37,9 +41,15 @@
 
     @queries = @matcher_execution_context.executor.call(&actual)
 
+    scale_factors = @queries.map(&:first)
     counts = @queries.map(&:last).map(&:size)
 
-    counts.max == counts.min
+    if scale_factors.max == scale_factors.min
```

Comment:
This is needed because `with_scale_factors` can accept equal values, such as `with_scale_factors(1, 1)`, which would cause division by 0.